### PR TITLE
Connect flow to Web

### PR DESCRIPTION
**Pull request by olavoasantos**
## Summary

This PR creates a reusable client class that standardizes communication with the dev-server. With it, this PR refactors the standalone app to make use of it.

### Setup
1. Run `make bootstrap`
2. Run `make run testdata/shopifile.yml` to start the server
3. On another terminal, run `yarn start` to run the dev console

### Checks
- [ ] All extensions defined in the `testdata/shopifile.yml` should be listed on the `http://localhost:3000/`
- [ ] While on `http://localhost:3000/`, the network tab should have an active WebSocket connection to `http://localhost:8000/`
- [ ] The appropriate events should be emitted to the WebSocket when you `focus` extensions; `unfocus` extensions; `hides` extensions; and `refreshes` extensions.
- [ ] Starting `ngrok` with a tunnel to `http://localhost:8000/`, by accessing `http://localhost:3000/?url=wss://{NGROK_TUNNEL}.ngrok.io/extensions` the localhost WebSocket connection should be finalized and a new connection should be started with the tunnel.

### CODE REVIEW

File: packages/dev-console/src/types.ts [hunk 1]
```diff
@@ -1,60 +1,42 @@
...
-export interface BaseAction<T extends string> {
-  type: T;
-}
-// @shopify/eslint-plugins doesn't allow any generic other than T
-// If this becomes a problem, consider modifying this rule
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export interface Action<T extends string = string, P = any> {
-  type: T;
-  payload: P;
+  }
```
Comment (alanthai) [inline packages/dev-console/src/types.ts:26]: I like the naming Outbound and Inbound events, but is the `Record<>` structure more readable than `ConsoleActions = ` unions? It feels less intuitive here.

File: packages/dev-console/src/ExtensionServerClient/APIClient.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export class APIClient implements ExtensionServer.API.Client {
+  constructor(public url: string) {}
+
+  async extensions(): Promise<ExtensionServer.API.ExtensionsResponse> {
+    const response = await fetch(`${this.url}/extensions`);
+    const dto: ExtensionServer.API.ExtensionsResponse = await response.json();
```
Comment (alanthai) [inline packages/dev-console/src/ExtensionServerClient/APIClient.ts:6]: what does dto stand for?

File: packages/dev-console/src/ExtensionServerClient/types.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
+/* eslint-disable @typescript-eslint/naming-convention, @typescript-eslint/no-namespace */
```
Comment (alanthai) [inline packages/dev-console/src/ExtensionServerClient/types.ts:1]: This file is a lot more stuff to take in for users. Let's add docs in this PR, too

File: packages/dev-console/src/ExtensionServerClient/APIClient.ts (hunk 2 above)
Comment (olavoasantos) [reply packages/dev-console/src/ExtensionServerClient/APIClient.ts:6]: DTO stands for `data transfer object`. It's a common term for an object that carries data but has no logic.

File: packages/dev-console/src/ExtensionServerClient/types.ts (hunk 3 above)
Comment (olavoasantos) [reply packages/dev-console/src/ExtensionServerClient/types.ts:1]: Ok. I'll add it to the README file on the kit. I feel it's good to keep these docblocks here as they'll serve as reference for users using them as they show up with the IDE's autocomplete popup.

**Pull request closed without merging**